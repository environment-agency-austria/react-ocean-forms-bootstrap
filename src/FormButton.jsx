import React from 'react';
import { Button } from 'reactstrap';
import { FormButton as BaseFormButton } from 'react-ocean-forms';

/**
 * Wrapper for a bootstrap button that will
 * automatically disable the button if the
 * form is busy
 */
function FormButton(props) {
  return (
    <BaseFormButton component={Button} {...props} />
  );
}

FormButton.displayName = 'FormButton';

export default FormButton;
