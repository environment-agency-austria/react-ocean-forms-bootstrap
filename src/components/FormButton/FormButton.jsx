/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import React from 'react';
import { Button } from 'reactstrap';
import { FormButton as BaseFormButton } from 'react-ocean-forms';

/**
 * Wrapper for a bootstrap button that will
 * automatically disable the button if the
 * form is busy
 */
function FormButton(props) {
  return (
    <BaseFormButton component={Button} {...props} />
  );
}

FormButton.displayName = 'FormButton';

export default FormButton;
