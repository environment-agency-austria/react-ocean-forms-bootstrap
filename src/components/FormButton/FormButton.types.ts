import {FormButton, PropsOf, Subtract } from 'react-ocean-forms';
import { ButtonProps } from 'reactstrap';

type TUpstreamButtonProps = ButtonProps & JSX.LibraryManagedAttributes<typeof FormButton, PropsOf<typeof FormButton>>;

/**
 * Props for the FormButton component
 */
export interface IFormButtonProps extends Subtract<TUpstreamButtonProps, { component: string }> { }
