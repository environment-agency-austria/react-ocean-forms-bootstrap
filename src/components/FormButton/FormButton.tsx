/**
 * Copyright (c) 2018-present, Umweltbundesamt GmbH
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */

import * as React from 'react';

import { FormButton as BaseFormButton } from 'react-ocean-forms';
import { Button } from 'reactstrap';

import { IFormButtonProps } from './FormButton.types';

/**
 * Wrapper for a bootstrap button that will
 * automatically disable the button if the
 * form is busy
 */
export class FormButton extends React.Component<IFormButtonProps> {
  public static displayName: string = 'FormButton';

  public render(): JSX.Element {
    return (
      <BaseFormButton component={Button} {...this.props} />
    );
  }
}
