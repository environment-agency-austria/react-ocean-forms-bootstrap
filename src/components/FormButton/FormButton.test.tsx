import React from 'react';

import { render } from '@testing-library/react'
import { Form } from 'react-ocean-forms';

import { FormButton } from './FormButton';

describe('<FormButton />', () => {
  it('should render without crashing', () => {
    const { asFragment, getByRole, getByText } = render(
      <Form>
        <FormButton>Mock button</FormButton>
      </Form>
    );

    expect(getByRole('button')).toBeInTheDocument();
    expect(getByText('Mock button')).toBeVisible();
    expect(asFragment()).toMatchSnapshot();
  });
});
