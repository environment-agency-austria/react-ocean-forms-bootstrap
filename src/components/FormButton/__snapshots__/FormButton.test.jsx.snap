// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormButton /> should render without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <FormButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "component": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "component": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getSnapshotBeforeUpdate": true,
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
